.github/scripts/metrics_checker.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -o pipefail
+
+# search for any "new" or modified metric emissions
+metrics_modified=$(git --no-pager diff HEAD origin/main | grep -i "SetGauge\|EmitKey\|IncrCounter\|AddSample\|MeasureSince\|UpdateFilter")
+# search for PR body or title metric references
+metrics_in_pr_body=$(echo "$PR_BODY" | grep "metric")
+metrics_in_pr_title=$(echo "$PR_TITLE" | grep "metric")
+
+# if there have been code changes to any metric or mention of metrics in the pull request body
+if [ "$metrics_modified" ] || [ "$metrics_in_pr_body" ] || [ "$metrics_in_pr_title" ]; then
+  # need to check if there are modifications to metrics_test
+  modified_metrics_test_file=$(git --no-pager diff --name-only HEAD "$(git merge-base HEAD "origin/main")" | grep -i "metrics_test")
+  if [ "$modified_metrics_test_file" ]; then
+    # 1 happy path: metrics_test has been modified bc we modified metrics behavior
+    echo "PR seems to modify metrics behavior. It seems it has modified agent/metrics_test.go as well."
```

Comment:
Should this use `$modified_metrics_test_file` instead of a hard-coded `agent/metrics_test.go`? This makes me wonder if we should be using a different technique to target the test file. For instance, this would break if we ever moved or renamed this file 🤔

Reply:
> Should this use `$modified_metrics_test_file` instead of a hard-code

oh yes actually this is a good point! I can add this in; the cmd will look at the diff and only `metrics_test` so this should be fine if the file moves.